client/history/client.go

```diff
@@ -804,44 +805,37 @@ func (c *clientImpl) GetReplicationMessages(
 		req.Tokens = append(req.Tokens, token)
 	}
 
-	var wg sync.WaitGroup
-	wg.Add(len(requestsByPeer))
+	g := &errgroup.Group{}
```

Comment:
I'm not 100% sure it's an appropriate choice here, but since any error is treated as fatal to the whole group, it may be worth using a `g, ctx := errgroup.WithContext(ctx)` - it'll cancel the context on any error, so concurrent requests will be stopped asap rather than attempt to complete.

_usually_ I think WithContext produces better behavior, but there are definitely exceptions.

Reply:
I did not use it here, because previous logic also would not cancel other requests. But yeah, adding WithContext might be worth here.